### ISSUE TITLE
sqlfluff duplicates lines with `'{{ "', '".join(var("foo")) }}'` in dbt

### ISSUE THREAD

**Issue by jan-benisek** [sqlfluff/sqlfluff#5096]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

We sometimes use dbt `var` to avoid duplicating common filters and parameters. 
However, in these cases, `sqlfluff` duplicates the code and adds blank space in the strings, which breaks my SQL.

### Expected Behaviour

I'd expect the line to be formatted, but not duplicated or with extra blank spaces added.

### Observed Behaviour

Clearly the SQL is broken after `fix`, see the example below.

### How to reproduce

This is example of the model:
```sql
select
    case
        when 'foo' in (
            '{{ "', '".join(var("backend_events_with_email_cs")) }}'
        ) then 'yes' else null
    end as my_column
```

The variable:
```
vars:
  my_variable:
    [
      "foo",
      "bar",
      "whatever",
    ]
```

I'd run the lint first:
- `sqlfluff lint --dialect 'redshift' path/to/model/foo.sql`

And getting the suggestions:
```bash
L:  14 | P:  13 | LT05 | Line is too long (125 > 80).
                       | [layout.long_lines]
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```

Then I'd fix the issues:
- `sqlfluff fix --dialect 'redshift' path/to/model/foo.sql`

And this is the result
```sql
select
    case
        when 'foo' in (
            '            {{ "', '".join(var("my_variable")) }}
{{ "', '".join(var("my_variable")) }}'
        ) then 'yes'
    end as my_column
```

Compiled code:
```sql
select
    case
        when 'foo' in (
            '            foo', 'bar', 'whatever
foo', 'bar', 'whatever'
        ) then 'yes'
    end as my_column
```

From where you can see that the row was duplicated and furthermore, it added spaces to the string, which effectively breaks my sql.

### Dialect

Redshift and dbt.

### Version

2.3.0

### Configuration

```
[sqlfluff]
verbose = 1
dialect = redshift
templater = jinja
output_line_length = 80
max_line_length = 80
sql_file_exts = .sql

# RF04: Allow keywords to be used in SQL.
# ST06: Don't reorder columns.
exclude_rules = RF04, ST06
warnings = none

[sqlfluff:templater:dbt]
project_dir =foo/

[sqlfluff:indentation]
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-indent-locations
indent_unit = space
tab_space_size = 4
indented_joins = False
indented_ctes = False
indented_using_on = False
indented_on_contents = True
indented_then = True
indented_then_contents = False
allow_implicit_indents = False
template_blocks_indent = True
trailing_comments = False

[sqlfluff:rules]
ignore_comment_lines = False

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
ignore_words = None
ignore_words_regex = None

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none

[sqlfluff:rules:references.consistent]
force_enable = True

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:rules:convention.terminator]
multiline_newline = True
require_final_semicolon = False

[sqlfluff:rules:convention.casting_style]
preferred_type_casting_style = consistent

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = trailing

[sqlfluff:layout:type:join_clause]
line_position = strict

[sqlfluff:rules:layout.long_lines]
ignore_comment_lines = False
ignore_comment_clauses = False
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by derik-roby** [comment]
I have the issue with sqlfluff and dbt-databricks. Did you find a workaround ?

**Comment by jan-benisek** [comment]
Not really, we refactored / `--noqa` these parts 😞

**Comment by kgl78** [comment]
I have the same issue when running `sqlfulff fix`. I'm using `dialect = snowflake` and `templater = dbt`.

Lines where I have: `{{ dbt_utils.star(from=ref('base_blackrock_aladdin__breakdown'), except=["benchmark_ticker", "run_date"]) }},`

get duplicated to 

`{{ dbt_utils.star(from=ref('base_blackrock_aladdin__breakdown'), except=["benchmark_ticker", "run_date"]) }}        {{ dbt_utils.star(from=ref('base_blackrock_aladdin__breakdown'), except=["benchmark_ticker", "run_date"]) }},`

It seems to be anywhere I have code that starts with ``{{ dbt_utils.star`

**Comment by alanmcruickshank** [comment]
I've confirmed this bug is still around and can recreate it with this query on default configuration:
```jinja
select
    '{{ "', '".join(["foo", "bar", "whatever", "whatever", "whatever", "whatever"]) }}'

```
